### PR TITLE
Fix reference parsing to support registry with port

### PR DESCRIPTION
**Pull request by sajayantony**
For the driver to work with a registry of the form `kind-registry:5000/myartifact:latest` the parsing needs to handle portl 

```
time="2023-04-17T23:28:35Z" level=info msg="Preparing to pull from remote repository: kind-registry:5000/github-ci"
time="2023-04-17T23:28:35Z" level=info msg="Plain http: true"
time="2023-04-17T23:28:35Z" level=info msg="Found digest: sha256:5d6742ff0b10c1196202765dafb43275259bcbdbd3868c19ba1d19476c088867 for latest"
time="2023-04-17T23:28:35Z" level=info msg="Pulling sha256:acb1ec674e686f4ba7a0e5c0ce1d41b6c2a5f5f1b9b9baca9c612f794faa3f8e, 0 of 1"
time="2023-04-17T23:28:35Z" level=info msg="OCI: Writing sha256:acb1ec674e686f4ba7a0e5c0ce1d41b6c2a5f5f1b9b9baca9c612f794faa3f8e to /pv_data/default/kind-registry:5000-github-ci-latest/container.sif"
time="2023-04-17T23:28:35Z" level=info msg="Oras artifact root: /pv_data/default/kind-registry:5000-github-ci-latest"
time="2023-04-17T23:28:35Z" level=info msg="Found artifact asset: container.sif"
time="2023-04-17T23:28:35Z" level=info msg="volume source directory:/pv_data/default/kind-registry:5000-github-ci-latest"
time="2023-04-17T23:28:35Z" level=info msg="volume target directory:/var/lib/kubelet/pods/b713b715-f675-410a-a3c0-7ea45b7bfe10/volumes/kubernetes.io~csi/oras-inline/mount"
time="2023-04-17T23:28:35Z" level=info msg="volume options:[ro]"
time="2023-04-17T23:28:35Z" level=info msg="BindMount - source: /pv_data/default/kind-registry:5000-github-ci-latest, target: /var/lib/kubelet/pods/b713b715-f675-410a-a3c0-7ea45b7bfe10/volumes/kubernetes.io~csi/oras-inline/mount, options: [ro]"
time="2023-04-17T23:28:35Z" level=info msg="mount -o bind /pv_data/default/kind-registry:5000-github-ci-latest /var/lib/kubelet/pods/b713b715-f675-410a-a3c0-7ea45b7bfe10/volumes/kubernetes.io~csi/oras-inline/mount"
time="2023-04-17T23:28:35Z" level=info msg="Successfully mounted /pv_data/default/kind-registry:5000-github-ci-latest to /var/lib/kubelet/pods/b713b715-f675-410a-a3c0-7ea45b7bfe10/volumes/kubernetes.io~csi/oras-inline/mount"
time="2023-04-17T23:28:35Z" level=info msg="Mount had no errors"
time="2023-04-17T23:28:35Z" level=info msg="returning NodePublishVolumeResponse"
```

### CODE REVIEW

File: pkg/oras/settings.go [hunk 1]
```diff
@@ -24,18 +26,33 @@ type orasSettings struct {
...
+// parseContainer URI into a reference(registry+repository) and a tag
+func (settings *orasSettings) parseReference(reference string) error {
+	artifactRef := strings.TrimPrefix(reference, "oras://")
 	tag := "latest"
-	if strings.Contains(container, ":") {
-		parts := strings.Split(container, ":")
-		container = parts[0]
-		tag = parts[1]
+
+	// if the refeence doesn't have a / then add a docker.io/library prefix
```
Comment (vsoch) [inline pkg/oras/settings.go]:
```suggestion
	// if the reference doesn't have a / then add a docker.io/library prefix
```

File: pkg/oras/settings.go [hunk 2]
```diff
@@ -24,18 +26,33 @@ type orasSettings struct {
...
+	if !strings.Contains(artifactRef, "/") {
+		artifactRef = "docker.io/library/" + artifactRef
+	}
+
+	ref, err := registry.ParseReference(artifactRef)
+	if err != nil {
+		return err
 	}
+
+	// Oras uses the ref to mean digest or tag
```
Comment (vsoch) [inline pkg/oras/settings.go:46]: That's not confusing or anything! :laughing: It continues to make me chuckle how overloaded these terms are!

File: pkg/oras/settings_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
+package oras
+
+import "testing"
```
Comment (vsoch) [inline pkg/oras/settings_test.go:3]: Nice! I think we need to add a command in the `Makefile` for these to run, and run it in the main.yaml CI?

Comment (sajayantony) [reply pkg/oras/settings_test.go:3]: Added to the test command.

File: pkg/oras/settings.go (hunk 2 above)
Comment (sajayantony) [reply pkg/oras/settings.go:46]: Long story (but this is because of how the OCI layout doesn't have a notion of of a tag similar to the image)

**Review by vsoch — APPROVED** [review]

**Pull request merged**